File: ios/AnyMind/PromptRecorder/Helpers/SyncStatus.swift
import SwiftUI

struct SyncIcon {
    let name: String
    let color: Color
    let label: String
}

enum SyncStatus {
    static func icon(updatedAt: Date, lastSyncAt: Date?, syncEnabled: Bool) -> SyncIcon {
        if !syncEnabled {
            return SyncIcon(name: "icloud.slash", color: .red, label: "Sync disabled")
        }
        guard let lastSyncAt else {
            return SyncIcon(name: "icloud", color: .gray, label: "Not synced")
        }
        if updatedAt > lastSyncAt {
            return SyncIcon(name: "icloud", color: .gray, label: "Not synced")
        }
        return SyncIcon(name: "icloud.fill", color: .blue, label: "Synced")
    }
}

File: Android/AnyMind/app/src/main/java/com/anymind/anymind/util/SyncStatus.kt
package com.anymind.anymind.util

import com.anymind.anymind.R
import java.time.Instant


data class SyncIcon(val iconRes: Int, val tintRes: Int, val label: String)

object SyncStatus {
    fun icon(updatedAt: Instant, lastSyncAt: Instant?, syncEnabled: Boolean): SyncIcon {
        if (!syncEnabled) {
            return SyncIcon(R.drawable.ic_cloud_off, R.color.sync_red, "Sync disabled")
        }
        if (lastSyncAt == null || updatedAt.isAfter(lastSyncAt)) {
            return SyncIcon(R.drawable.ic_cloud, R.color.sync_gray, "Not synced")
        }
        return SyncIcon(R.drawable.ic_cloud_done, R.color.sync_blue, "Synced")
    }
}
